src/code_800AF9B0.c

```diff
@@ -3034,67 +2958,59 @@ void func_800B4A9C(s32 arg0) {
 GLOBAL_ASM("asm/non_matchings/code_800AF9B0/func_800B4A9C.s")
 #endif
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-? func_800B4820(); // extern
-s32 func_800B492C(); // extern
-s32 func_800B49E4(); // extern
-? func_800B5948(); // extern
-u8 func_800B59F4(); // extern
-u8 func_800B5AAC(); // extern
-s32 func_800B5AD8(); // extern
-extern ? D_8018EB90;
-extern u8 D_8018ED10;
-extern u8 D_8018ED11;
-extern u8 D_8018ED12;
-extern u8 D_8018ED13;
-extern u8 gSaveDataSoundMode;
-extern u8 D_8018ED16;
-extern u8 D_8018ED17;
-extern u8 D_8018ED88;
-extern u8 D_8018ED89;
-extern u8 D_8018ED8A;
-extern u8 D_8018ED8B;
-extern u8 gSaveDataSoundModeBackup;
-extern OSMesgQueue gSIEventMesgQueue;
-
-void func_800B4CB4(void) {
-    if ((D_8018ED16 != func_800B492C()) || (D_8018ED17 != func_800B49E4())) {
-        func_800B4820();
-        if (func_800B5AD8() == 0) {
+void validate_save_data(void) {
+    if ((D_8018ED16 != (compute_save_data_checksum_1() & 0xFFFFFFFF)) || (D_8018ED17 != (compute_save_data_checksum_2() & 0xFFFFFFFF))) {
+        
+        reset_save_data_grand_prix_points_and_sound_mode();
+        
+        if (validate_save_data_checksum_backup() == 0) {
             D_8018ED10 = D_8018ED88;
+            if ((D_8018ED16 && D_8018ED16) && D_8018ED16)
+            {
+            }
+
             D_8018ED11 = D_8018ED89;
+            if ((D_8018ED16 && D_8018ED16) && D_8018ED16)
+            {
+            }
+
             D_8018ED12 = D_8018ED8A;
+            if ((D_8018ED16 && D_8018ED16) && D_8018ED16)
+            {
+            }
+
             D_8018ED13 = D_8018ED8B;
+            if ((D_8018ED16 && D_8018ED16) && D_8018ED16)
+            {
+            }
+
             gSaveDataSoundMode = gSaveDataSoundModeBackup;
-            D_8018ED16 = func_800B59F4();
-            D_8018ED17 = func_800B5AAC();
-            osEepromLongWrite(&gSIEventMesgQueue, ((&D_8018ED10 - &D_8018EB90) >> 3) & 0xFF, &D_8018ED10, 8);
+            D_8018ED16 = compute_save_data_checksum_backup_1();
+            D_8018ED17 = compute_save_data_checksum_backup_2();
+            osEepromLongWrite(&gSIEventMesgQueue, (((u32) &D_8018ED10 - (u32) &D_8018EB90) >> 3), &D_8018ED10, 8);
         }
-        func_800B5948();
+        update_save_data_backup();
         return;
     }
-    if (func_800B5AD8() != 0) {
-        func_800B5948();
+
+    if (validate_save_data_checksum_backup() != 0) {
+        update_save_data_backup();
     }
 }
-#else
-GLOBAL_ASM("asm/non_matchings/code_800AF9B0/func_800B4CB4.s")
-#endif
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-void func_800B4DCC(void *arg0, u32 arg1, s32 arg2) {
-    u32 temp_t8;
+void write_time_trial_record(u8 *timeTrialRecord, u32 time, s32 characterId)
```

Comment:
Maybe populate_time_trial_record would be a better name? Avoid confusion since no eeprom writes are called in this function

Reply:
I agree